File: src/19. Remove Nth Node From End of List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
function removeNthFromEnd(head, n) {
  const dummy = new ListNode(-1, head)
  let left = dummy,
    right = dummy
  while (n) {
    right = right.next
    n--
  }
  while (right.next) {
    left = left.next
    right = right.next
  }
  const deleted = left.next
  left.next = deleted.next
  deleted.next = null
  return dummy.next
}

File: src/19. Remove Nth Node From End of List.py
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        dummy = ListNode(-1, head)
        left = right = dummy
        while n:
            right = right.next
            n -= 1
        while right.next:
            left = left.next
            right = right.next
        deleted = left.next
        left.next = deleted.next
        deleted.next = None
        return dummy.next